### ISSUE TITLE
Improve torrent graph UI

### ISSUE THREAD

**Issue by feross** [feross/webtorrent-www#73]
- [ ] change node color depending on seed/leech state
- [ ] show transfer rate between nodes (by changing thickness of line?)
- [ ] show additional peer info (client name, version, country) when clicking on peer

**Comment by roccomuso** [comment]
I could help with something but we should first split it in a separete module: #63

**Comment by feross** [comment]
@roccomuso If you want to send improvements here first, I can publish it a separate module in a few days.

**Comment by roccomuso** [comment]
@feross I'm working on this features.
Here https://github.com/roccomuso/webtorrent-www/commit/2bbe5b02b3d2016ef2f7833cb440aece502b3476 the change of node color depending on seed/leech state. Is <code>graph.seed('id', true/false);</code> acceptable as API?

I've noticed also that the *choke* and *unchoke* methods need to be completed. Choking acts on the link and not on the node itself right? We could change the line color or something like that?

**Comment by feross** [comment]
I think that API works!

I think making the line more transparent when we're choked would be good.

**Comment by roccomuso** [comment]
ok thanks for the tip, I'm on it right now... ;)

By the way I'm introducing a new dependency: https://github.com/avoidwork/filesize.js
It's a few lines browserified but provides a pratical way to show and work with transfer rate and byte size conversion.

**Comment by feross** [comment]
@roccomuso We use `prettier-bytes` throughout the webtorrent projects. Please use that to avoid sending extra code to people's browsers 👍

**Comment by roccomuso** [comment]
@feross Alright, now using <code>prettier-bytes</code>.
Check out the new feature: *show transfer rate between nodes (by changing thickness of line)* - https://github.com/roccomuso/webtorrent-www/commit/0e67935bf0186ff7cc00ea4ef5cf488cbe7e7b00
The API is: <code>graph.rate('Peer1', 'Peer2', 256 * 1000);</code>, where the last param is a number expressed in bytes.

On the top of the lib in the COLORS variable I've defined some properties required to "map" the transfer rate to the line thickness.

**Comment by feross** [comment]
Screenshot?

**Comment by roccomuso** [comment]
![gifrecord_2016-09-19_114756](https://cloud.githubusercontent.com/assets/6730217/18628182/383d7a7c-7e5f-11e6-8444-74a184fc121a.gif)

@feross As you can see the link gets thicker and thinner, changing every second his transfer rate to:
1) 150 KB/s.
2) 500 KB/s.
3) 2.5 MB/s.
4) 5 MB/s.
5) 2.5 MB/s.
6) 1 MB/s.

**Comment by feross** [comment]
That's nice. Can you send a PR?

**Comment by roccomuso** [comment]
@feross yes sure. Couldn't be first splitted it in a separate module? So that I can write also the API documentation in a README.md.

**Comment by feross** [comment]
Sure, I'll do that now and add you as GitHub collaborator and npm owner.

**Comment by roccomuso** [comment]
@feross thanks, let me know ;)

**Comment by feross** [comment]
Okay, it's published as `p2p-graph`: https://github.com/feross/p2p-graph I added you to the repo/package.

**Comment by feross** [comment]
Here's a PR to make `webtorrent-www` use `p2p-graph`: https://github.com/feross/webtorrent-www/pull/105

**Comment by roccomuso** [comment]
ok, I think you can merge the PR. I'm gonna push directly on the p2p-graph repo.

**Comment by feross** [comment]
@roccomuso Nice. Please still send PRs to the `p2p-graph` repo so there's a chance for me and others to give feedback before merging.

**Comment by roccomuso** [comment]
@feross ok cool!
